WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -217,13 +222,46 @@ public void onSiteRemoved(OnSiteRemoved event) {
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onSiteChanged(OnSiteChanged event) {
-        mDebouncer.debounce(Void.class, new Runnable() {
-            @Override public void run() {
-                if (!isFinishing()) {
-                    getAdapter().loadSites();
-                }
+        if (mSwipeToRefreshHelper.isRefreshing()) {
+            mSwipeToRefreshHelper.setRefreshing(false);
+            if (!isFinishing()) {
+                getAdapter().loadSites();
             }
-        }, 200, TimeUnit.MILLISECONDS);
+        } else {
+            mDebouncer.debounce(Void.class, new Runnable() {
+                @Override
+                public void run() {
+                    if (!isFinishing()) {
+                        getAdapter().loadSites();
+                    }
+                }
+            }, 200, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void initSwipeToRefreshHelper(View view) {
+        if (view == null) {
+            return;
+        }
+        mSwipeToRefreshHelper = new SwipeToRefreshHelper(
+                this,
+                (CustomSwipeRefreshLayout) view.findViewById(R.id.ptr_layout),
+                new SwipeToRefreshHelper.RefreshListener() {
+                    @Override
+                    public void onRefreshStarted() {
+                        if (isFinishing()) {
+                            return;
+                        }
+                        if (!NetworkUtils.checkConnection(SitePickerActivity.this)) {
+                            mSwipeToRefreshHelper.setRefreshing(false);
+                            ToastUtils.showToast(SitePickerActivity.this, R.string.site_picker_no_network_refresh_error);
```

Comment:
Removed the extra toast and corresponding error string in 22286aaad0d2a969438446e2ac06eeebab3205a2